iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingViewIndex.java

```diff
@@ -51,10 +55,52 @@ MergingView getView() {
         return networks.stream();
     }
 
+    boolean containsId(final String id) {
+        for (Network n : networks) {
+            if (Objects.nonNull(n.getIdentifiable(id))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    String getUniqueId(final String baseId) {
+        String checkedBaseId;
+        if (baseId != null && baseId.length() > 0) {
+            if (!containsId(baseId)) {
+                return baseId;
+            }
+            checkedBaseId = baseId;
+        } else {
+            checkedBaseId = "autoid";
+        }
+        String uniqueId;
+        int i = 0;
+        do {
+            uniqueId = checkedBaseId + '#' + i++;
+        } while (i < Integer.MAX_VALUE && containsId(uniqueId));
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("Object '{}' is not unique, rename to '{}'", baseId, uniqueId);
+        }
+        return uniqueId;
+    }
+
+    void checkUniqueIds(final Network other, final MergingViewIndex index) {
+        // check mergeability
+        final Collection<String> otherIds = other.getIdentifiables().stream()
+                .map(Identifiable::getId)
+                .collect(Collectors.toSet());
+        index.getIdentifiableStream().map(Identifiable::getId).forEach(id -> {
+            if (otherIds.contains(id)) {
+                throw new PowsyblException("The object '" + id + "' already exists into merging view");
+            }
+        });
+    }
```

Comment:
I think it's not really efficient if you want to ensure that this ID already exists: your `containsId` method is really better.

Reply:
Indeed, this method is not used yet.
I will delete it